src/main/kotlin/com/openlattice/transporter/TransporterQueries.kt

```diff
@@ -0,0 +1,145 @@
+            rs.next()
+        }
+    }
+}
+
+val pkCols = listOf(ENTITY_SET_ID, ID_VALUE)
+val pk = pkCols.joinToString(", ") { it.name }
+val ids = "$pk,${ORIGIN_ID.name}"
+
+/**
+ * column bindings are
+ * 1 - partitions array
+ * 2 - entity set ids array
+ *
+ */
+fun updateOneBatchForProperty(destTable: String, propId: UUID, propCol: TransporterColumn): String {
+    val srcCol = propCol.srcCol
+    val destCol = propCol.destColName
+    val dataCol = "CASE WHEN VERSION > 0 THEN $srcCol else null END as $destCol"
+
+    val updateLastPropagate = "UPDATE ${PostgresTable.DATA.name} " +
+            "SET ${transportTimestampColumn.name} = ${DataTables.LAST_WRITE.name} " +
+            "WHERE ${PARTITION.name} = ANY(?) " +
+            " AND ${ENTITY_SET_ID.name} = ANY(?) " +
+            " AND ${PROPERTY_TYPE_ID.name} = '${propId}' " +
+            " AND ${DataTables.LAST_WRITE.name} > ${transportTimestampColumn.name} " +
+            "RETURNING $ids,${VERSION.name},$dataCol"
+
+    // this seems unnecessary but it's good to be sure
+    val createMissingRows = "INSERT INTO $destTable " +
+            "($ids) " +
+            "SELECT $ids " +
+            "FROM src " +
+            "WHERE ${VERSION.name} > 0 " +
+            "ON CONFLICT ($pk) DO NOTHING"
+    val modifyDestination = "UPDATE $destTable t " +
+            "SET $destCol = src.$destCol " +
+            "FROM src " +
+            "WHERE " + pkCols.joinToString(" AND ") { "t." + it.name + " = src." + it.name}
+
+    return "WITH src as (${updateLastPropagate}), inserted as ($createMissingRows) $modifyDestination"
+}
+
+/**
+ * column bindings are
+ * 1 - partitions array
+ * 2 - entity set ids array
+ *
+ */
+fun updateIdsForEntitySets(destTable: String): String {
+    val updateLastPropagate = "UPDATE ${PostgresTable.IDS.name} " +
+            "SET ${transportTimestampColumn.name} = ${DataTables.LAST_WRITE.name} " +
+            "WHERE ${PARTITION.name} = ANY(?) " +
+            " AND ${ENTITY_SET_ID.name} = ANY(?) " +
+            " AND ${DataTables.LAST_WRITE.name} > ${transportTimestampColumn.name} " +
+            "RETURNING ${ENTITY_SET_ID.name}, ${ID_VALUE.name},${LINKING_ID.name},${VERSION.name}"
+    val createMissingRows = "INSERT INTO $destTable ($pk) " +
+            "SELECT $pk " +
+            "FROM src " +
+            "WHERE ${VERSION.name} > 0 " +
+            "ON CONFLICT ($pk) DO NOTHING"
+    val createMissingLinkedRows = "INSERT INTO $destTable ($pk,${ORIGIN_ID.name}) " +
+            "SELECT ${ENTITY_SET_ID.name}, " +
+            " ${LINKING_ID.name} AS ${ID_VALUE.name}, " +
+            " ${ID_VALUE.name} AS ${ORIGIN_ID.name} " +
+            "FROM src " +
+            "WHERE ${VERSION.name} > 0 " +
+            " AND ${LINKING_ID.name} IS NOT NULL " +
+            "ON CONFLICT ($pk) DO NOTHING"
+    val deleteRows = "DELETE FROM $destTable t " +
+            "USING src " +
+            "WHERE src.${VERSION.name} <= 0 " +
```

Comment:
I'm not sure what the current state of this is, but there should theoretically be a job that finds ids with `version = 0` (i.e. hard deleted), and goes and does all the cleanup required to make sure that entity is removed everywhere (data/ids/e table, materialized views, elasticsearch, linking clusters, etc). Seems like a good idea to include <= 0 in case this gets hit first, but was there any code added to a background hard-delete-cleanup job to make sure the rows will for sure be removed, even if that background job is executed before this one? (if such a job exists, lol)

Reply:
I think the plan is here, that that job won't run on entries, where `transportTimestampColumn < last_write`.. but both solutions would be fine I think.

Reply:
Does that affect whether or not we need code in a background deletion task?

Assuming we have a cleanup job that deletes rows from the data table with version = 0 -- if that background job finds a hard-deleted row before this background job finds it, then unless that one also cleans up the transported one, that row won't ever be deleted.

Reply:
`BackgroundIndexedEntitiesDeletionService.selectDeletedIds` is where we select hard deleted entity ids for removal. I've updated it here: openlattice/indexer#117